### PR TITLE
Add docs & tests for Table.set_format

### PR DESCRIPTION
**Pull request by hmstepanek**
- [x] Add docs & tests for Table.set_format

This is a partial fix for [Add documentation](https://github.com/data-8/datascience/issues/163) as it adds documentation for Table.set_format.

### CODE REVIEW

File: datascience/tables.py [hunk 1]
```diff
@@ -3052,7 +3219,6 @@ def draw_hist(values_dict):
                         #There's a grouping in place but we're only plotting one column's values
                         label_not_grouped = [l for l in self.labels if l != group][0]
                         axis.set_xlabel(label_not_grouped + x_unit, fontsize=16)
-                    else:
```
Comment (adnanhemani) [inline datascience/tables.py:3055]: Typo?

File: datascience/tables.py [hunk 2]
```diff
@@ -514,7 +514,174 @@ def last(self, label):
     ############
 
     def set_format(self, column_or_columns, formatter):
-        """Set the format of a column."""
+        """
```
Comment (adnanhemani) [inline datascience/tables.py:517]: Could you please add an example of one of the built-in formatter classes near the top of this docstring? Most of the users of our library, I image, would be using these quite frequently.

**Review by adnanhemani — CHANGES_REQUESTED** [review]
Thanks for this! Just a few comments

**Comment by hmstepanek** [comment]
I've made the first example use the builtin CurrencyFormatter and also made the custom formatter examples more relatable since the builtin CurrencyFormatter would not be able to handle these cases (since it always prepends the symbol). Also fixed the typo.

**Review by adnanhemani — APPROVED** [review]
Nice work - thanks for making this documentation so detailed!

**Pull request merged**